make/autoconf/jdk-options.m4

```diff
@@ -438,6 +438,14 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_ADDRESS_SANITIZER],
           # It's harmless to be suppressed in clang as well.
           ASAN_CFLAGS="-fsanitize=address -Wno-stringop-truncation -fno-omit-frame-pointer -fno-common -DADDRESS_SANITIZER"
           ASAN_LDFLAGS="-fsanitize=address"
+          # detect_stack_use_after_return is using a fake stack and thus JDK would crash:
+          # assert(_thread->is_in_live_stack((address)this)) failed: not on stack?
```

Comment:
Proposal:

|detect_stack_use_after_return causes ASAN to offload stack-local variables to c-heap and therefore breaks assumptions in hotspot that rely on data (e.g. Marks) to live in thread stacks."